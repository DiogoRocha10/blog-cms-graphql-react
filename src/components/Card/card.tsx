import styles from "./card.module.css";
import cn from "clsx";

interface CardProps {
  title: string;
  children: React.ReactNode;
  className?: string;
  description?: string;
  icon?: React.ReactNode;
}

export const Card = ({
  title,
  className,
  children,
  description,
  icon,
}: CardProps) => {
  return (
    <a className={cn(className, styles["-card--container"])}>
      <div className={styles["-card--content"]}>{children}</div>
      <div className={styles["-card--text-container"]}>
        <strong className={styles["-card--title"]}>{title}</strong>
        <p className={styles["-card--description"]}>{description}</p>
      </div>
      <div className={styles["-card--icon"]}>{icon}</div>
    </a>
  );
};
