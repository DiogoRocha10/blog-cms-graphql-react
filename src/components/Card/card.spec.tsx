import { render } from "@testing-library/react";
import { Card } from "./card";
import renderer from "react-test-renderer";

it("text", () => {
  const { getByText } = render(<Card title="Card" />);
  expect(getByText("Card")).toBeTruthy();
});

it("renders correctly", () => {
  const tree = renderer.create(<Card title="Card" />).toJSON();
  expect(tree).toMatchSnapshot();
});
