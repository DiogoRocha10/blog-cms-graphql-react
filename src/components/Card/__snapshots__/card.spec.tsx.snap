// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="-card--container"
>
  <div
    className="-card--content"
  />
  <div
    className="-card--text-container"
  >
    <strong
      className="-card--title"
    >
      Card
    </strong>
    <p
      className="-card--description"
    />
  </div>
  <div
    className="-card--icon"
  />
</a>
`;
